src/main/scala/intellij/haskell/action/DownloadCabalConfigFile.scala

```diff
@@ -0,0 +1,17 @@
+package intellij.haskell.action
+
+import com.intellij.openapi.actionSystem.{AnAction, AnActionEvent}
+import intellij.haskell.external.component.HaskellComponentsManager.refreshGlobalProjectInfo
+import intellij.haskell.external.component.PathComponent
+
```

Comment:
> Restarting the repls will also invaliYes, currently the restart repl action will also trigger the checking and downloading of cabal config file here, but restart repl will take a relatively long time, and the completion of package names did not reply on other things, so maybe extract the cabal file downloading action is more flexible for users to use?date GlobalCaches.

When package completion is requested and there is no cabal.config file it has to be downloaded. Because of the case user changes Stackage LTS release, every time project is opened the cabal.config has to be downloaded. Downloading library sources is only done at request and when creating project because it takes some time.